Added '%matplotlib agg' option.

teoguso opened the pull request:
Behaves quite like inline, but it does not
display figures automatically.
Closes  #9389. 
cc @akhmerov @fperez 

Done at the jupyter sprint at EuroScipy 2016.

teoguso committed:
Added '%matplotlib agg' option.

Behaves quite like inline, but it does not
display figures automatically.

minrk commented on IPython/core/pylabtools.py:
```diff
@@ -261,6 +262,8 @@ def find_gui_and_backend(gui=None, gui_select=None):
         # select backend based on requested gui
         backend = backends[gui]
+        if gui == 'agg':
+            gui = None
```
This is needed because of an error in ipykernel, where it doesn't recognize `'agg'` as a GUI and it errors if it doesn't know what GUI to use. None indicates "no GUI", which is correct in principle, but there may be a better place to put that information.

We could modify something either here or in ipykernel to make it easier to specify "this backend needs no GUI integration", which would be true for nbagg, inline, agg.

akhmerov commented:
Thanks! Do you mind also updating the docstring of the magic so that it's know that this backend exists?

willingc commented on IPython/core/pylabtools.py:
```diff
@@ -261,6 +262,8 @@ def find_gui_and_backend(gui=None, gui_select=None):
     if gui and gui != 'auto':
         # select backend based on requested gui
         backend = backends[gui]
+        if gui == 'agg':
```
Since `'agg'` is an addition to the function, it would be helpful to add it to the function's docstring parameters for `gui`.

> We could modify something either here or in ipykernel to make it easier to specify "this backend needs no GUI integration", which would be true for nbagg, inline, agg.

Good suggestion.

teoguso committed:
Updated docstring for agg backend and gui

Updated the list of possible values for gui in the
docstring, according to my previous commit.

minrk commented:
Thanks, @teoguso!
